Remove strumpack, make 64bit the default

milljm opened the pull request:
When building external library 'strumpack' for PETSc, strumpack ignores compiler
flags used during PETSc configure. This causing link warnings with fortran during
build times, for libraries depending on PETSc.

moosebuild commented:
All [jobs](https://civet.inl.gov/event/42492/) on 063f4f0 : canceled by @milljm

this needs to be a new PR

Commit:
Remove Strumpack

Strumpack is linking to system fortran libraries even while supplying
different build flags to PETSc's configure system. Therefore we can not
support strumprack in our redistributable.
